src/control/system/database.go

```diff
@@ -468,27 +468,29 @@ func (db *Database) GroupMap() (*GroupMap, error) {
 	return gm, nil
 }
 
-// ReplicaRanks returns the set of ranks associated with MS replicas.
-func (db *Database) ReplicaRanks() (*GroupMap, error) {
+// GroupMapWithReplicaRanks returns the latest system group map with
+// the set of ranks associated with MS replicas.
+func (db *Database) GroupMapWithReplicaRanks() (*GroupMap, []Rank, error) {
 	if err := db.CheckReplica(); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	db.data.RLock()
 	defer db.data.RUnlock()
 
 	gm := newGroupMap(db.data.MapVersion)
-	for _, srv := range db.filterMembers(AvailableMemberFilter) {
-		if !db.isReplica(srv.Addr) {
-			continue
+	var replicaRanks []Rank
+	for _, srv := range db.data.Members.Ranks {
+		if srv.state&AvailableMemberFilter > 0 && db.isReplica(srv.Addr) {
+			replicaRanks = append(replicaRanks, srv.Rank)
 		}
```

Comment:
I recall that's my first attempt, and there was some concern, and I compromised by using the current method. But I don't recall why. Let me try filterMembers again...

Reply:
Now I recall: Because Database.GroupMap does not perform this filtering, while I'd like to keep the resulting GroupMap and ReplicaRanks unchanged, not knowing if there'd be any subtle differences.